src/components/atoms/markdown/CodeFence.tsx

```diff
@@ -53,6 +79,16 @@ const CodeFence = (props: React.HTMLProps<HTMLPreElement> = {}) => {
         >
           <Icon path={mdiContentCopy} />
         </CodeFenceCopyButton>
+        {rawContent && <CodeFenceSaveButton
+          onClick={() => {
+            var blob = new Blob([rawContent], { type: "text/plain;charset=utf-8" });
+            saveAs(blob, "hello world.txt");
```

Comment:
I think we can guess extension and mime type from the language of code blocks!

Reply:
I can't do it. If you can, why need add the template like `js css html` after **```**
mime type does not matter, it is always text
All I can do is get ext from className